### ISSUE TITLE
Feature Request: "training" argument for contrib.rnn.DropoutWrapper like the one in tf.layers.dropout

### ISSUE THREAD

**Issue by nelson-liu** [tensorflow/tensorflow#9775]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.0.4
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: 1.1.0

### Feature Request: "training" argument for contrib.rnn.DropoutWrapper for applying dropout depending on train/inference phase.

In [`tf.layers.dropout`](https://www.tensorflow.org/api_docs/python/tf/layers/dropout), the `training` parameter is a handy setting that lets you apply dropout depending on whether the model is training or doing inference. It's very convenient to be able to pass a boolean to the model placeholder and have it automatically do the right thing when it comes to dropout.

Unfortunately, [`tf.contrib.rnn.DropoutWrapper`](https://www.tensorflow.org/api_docs/python/tf/contrib/rnn/DropoutWrapper) does not have this same parameter, and I think it would greatly benefit from it. This is a feature request for it.

I tried implementing it myself with `tf.cond` and either returning the dropped-out outputs/states or the untouched ones, but I couldn't figure out how to share the variables between them in the cond.

**Comment by girving** [comment]
@ebrevdo This seems like a reasonable request.  Can you comment on the best way to add it?

**Comment by sogangori** [comment]
keep_prob = tf.cond(dropOut,lambda:tf.constant(0.9), lambda:tf.constant(1.0))
cells = rnn.DropoutWrapper(cells, output_keep_prob=keep_prob)

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activityand the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by ebrevdo** [comment]
Does "training" need to be a tensor, or can it be a regular Python bool always?  If so, you don't need the cond() call.

**Comment by songmeixu** [comment]
I understand your confusion. Use a python bool for is_training, it cannot be in graph; while use a tensor.bool for is_training, you need something like tf.is_true(is_training) , but tf without it.

The problem can be solve by:
`
is_training = tf.placeholder_with_default(False, shape=(), name='is_training')
basic_cell = tf.contrib.rnn.BasicRNNCell(num_units=n_neurons)
keep_prob = tf.cond(is_training, lambda:tf.constant(0.7), lambda:tf.constant(1.0))
basic_cell = tf.contrib.rnn.DropoutWrapper(basic_cell, input_keep_prob=keep_prob)
outputs, states = tf.nn.dynamic_rnn(basic_cell, X, dtype=tf.float32)

# trainging
sess.run(outputs, feed_dict={is_training: True})
`

**Comment by sdalibor** [comment]
Would this approach cause a dropout scaling issue? As I have understood scaling factor is applied to the weights during training phase according to the dropout factor, should it be OK to just change the dropout factor during inference phase?

**Comment by dynamicwebpaige** [comment]
Closing this issue, as `tf.contrib` is deprecated in TensorFlow 2.0, and has never been officially supported. Please reopen if the issue is still present in other TF RNN implementations.
